end_users/setup/README.md

```diff
@@ -2,9 +2,72 @@
 
 ## Table of Contents
 
+* [Robots](#robots)
+* [Webhook](#webhook)
+* [Hoist Configuration](#hoist-configuration)
+  * [Zuul Job Configuration](#zuul-job-configuration)
+  * [Zuul Layout Configuration](#zuul-layour-configuration)
 * [Merge Options](#merge-options)
   * [Setting the Correct Merge Option](#setting-the-correct-merge-option)
 
+## Robots
+
+The first step to getting BonnyCI into your repository is to give the [robots](https://github.com/orgs/BonnyCI/teams/robots) team write access. This is so BonnyCI can automatically merge pull requests that pass all testing queues. Navigate to your repository's settings page, click on the `Collaborators & teams` section in the left-hand menu, and click `Add a team:` in the dropdown menu under the `Teams` section.
+
+For more on organizations and teams, check out the [Github documentation](https://help.github.com/enterprise/2.8/admin/guides/user-management/organizations-and-teams/).
+
+## Webhook
+
+The next step to getting BonnyCI into your repository is to add the BonnyCI webhook. This is to notify BonnyCI of events taking place in your repository, such as newly opened pull requests. Navigate to your repository's settings page, click on the `Webhooks` section in the left-hand menu, and click the `Add webhook` button to the right. Paste the following link into the `Payload URL` item:
+
+```webhook
+http://zuul.bonnyci.org:8001/connection/github/payload
+```
+
+Ensure the `Content Type` dropdown is set to `application/json`, the `Send me everything.` radio button is selected, and the `Active` box is checked. Once this is done, click the green `Add webhook` button to finish. Here is a sample of what the settings will look like:
+
+![Correct Webhook Configuration](../../misc/images/BonnyCIWebhook.png)
+
+## Hoist Configuration
+
+We'll need a few pieces of configuration to be added to [hoist](https://github.com/BonnyCI/hoist) in order for your project to gain the necessary zuul testing queues. Create a fork of BonnyCI, clone it, check out a branch, write these files, and commit your changes to a branch in your fork. From there, open a pull request to hoist.
+
+### Zuul Job Configuration
+
+This file will add a definition for what tests to run on your repository with BonnyCI. Place it in `hoist/roles/zuul/files/jobs/repo_name.yaml`. Here is a sample configuration:
+
+```YAML
+- job:
+    name: repo_name
```

Comment:
It's more important to explain that this is a unique job name than to tie it to the repo name. I think `unique_job_name` would be better than `repo_name` so that it is clear in the next section that they match.